src/Dialog/Dialog.test.tsx

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2015 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
+ */
+
+import React from 'react';
+import { fireEvent, screen } from '@testing-library/react';
+
+import render from '../../test/testrenderer';
+import {
+    ConfirmationDialog,
+    Dialog,
+    DialogButton,
+    ErrorDialog,
+    InfoDialog,
+} from './Dialog';
+
+const noop = jest.fn();
+describe('Dialog', () => {
+    const dialog = (isVisible = true) => (
+        <Dialog isVisible={isVisible}>
+            <Dialog.Header title="Test Title" />
+            <Dialog.Body>Test Text</Dialog.Body>
+            <Dialog.Footer>
+                <DialogButton onClick={noop}>Close</DialogButton>
+            </Dialog.Footer>
+        </Dialog>
+    );
+
+    describe('is visible when it', () => {
+        it('is rendered when visible', () => {
+            render(dialog());
+            expect(screen.getByRole('dialog')).toBeInTheDocument();
+            test;
+        });
+
+        it('is not rendered when invisible', () => {
```

Comment:
'Dialog is visible when it is not rendered when invisible'

=> I suggest, remove the middle describe, rewrite the descriptions
e.g. ' Dialog is immediately visible by default' (The previous it())
and 'Dialog is hidden when you provide argument visible=true'

Please, if you don't mind, replace `it` with `test`. It's not important, but we have discussed earlier, and `test`, at least for me, makes more sense then the old `it` token.

Reply:
Fixed